packages/roosterjs-editor-dom/lib/edit/pasteTable.ts

```diff
@@ -0,0 +1,74 @@
+import findClosestElementAncestor from '../utils/findClosestElementAncestor';
+import getTagOfNode from '../utils/getTagOfNode';
+import toArray from '../jsUtils/toArray';
+import VTable from '../table/VTable';
+import { NodePosition, NodeType, TableOperation } from 'roosterjs-editor-types';
+
+/**
+ * @internal
+ * @param root The table to paste into
+ * @param nodeToInsert A Node containing the table to be inserted
+ * @param position The position to paste the table
+ */
+export default function pasteTable(
+    root: HTMLElement,
+    nodeToInsert: Node,
+    position: NodePosition,
+    range: Range
+) {
+    if (
+        (nodeToInsert.childNodes.length == 1 &&
+            getTagOfNode(nodeToInsert.childNodes[0]) == 'TABLE') ||
+        (nodeToInsert.childNodes.length == 2 &&
+            getTagOfNode(nodeToInsert.childNodes[0]) == 'TABLE' &&
+            getTagOfNode(nodeToInsert.childNodes[1]) == 'BR')
+    ) {
+        let rootNodeToInsert: Node | null = nodeToInsert;
+
+        if (rootNodeToInsert.nodeType == NodeType.DocumentFragment) {
+            let rootNodes = toArray(rootNodeToInsert.childNodes).filter(
+                (n: ChildNode) => getTagOfNode(n) != 'BR'
+            );
+            rootNodeToInsert = rootNodes.length == 1 ? rootNodes[0] : null;
+        }
+
+        let tdNode = findClosestElementAncestor(position.node, root, 'TD,TH');
```

Comment:
this is for TD in existing document, right? Let's rename it to make it easy to understand